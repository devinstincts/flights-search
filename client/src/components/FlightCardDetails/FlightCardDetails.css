.flightCardDetailsContainer {
    background: #fff;
    border-radius: 4px;
    padding: 8px;
    margin-bottom: 15px;
}
.card-footer-container {
    display: flex;
    justify-content: space-between;
    align-items: flex-end;
}
.card-footer-amount {
    font-size: 20px;
    font-weight: bold;
}
.card-footer-agent {
    color: #B0B0B1;
}
button.card-select-btn {
    padding: 6px 31px;
    background: #00ca6e;
    border-radius: 25px;
    color: #fff;
    border: 0;
    font-weight: bold;
    height: 35px;
    font-size: 14px;
}
