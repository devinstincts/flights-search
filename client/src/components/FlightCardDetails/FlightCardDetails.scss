@import '~bpk-mixins';

.FlightCardDetails {
    margin: 0 0 20px 0;
    background: #fff;
    display: flex;
    justify-content: space-between;

    &__parent-container {
        background: #fff;
        border-radius: 10px;
        padding: 10px;
        margin: 15px 0px 5px 0px;
    }

    &__origin-details, 
    &__destination-details {
        p {
            margin: 0;
            font-size: 14px;
            line-height: normal;

            &:first-child {
                margin: 0 0 10px;
            }
        }
    }

    &__origin-details {
        margin-right: 15px;
    }

    &__duration-details {
        p {
            font-size: 12px;
            margin: 0;
            line-height: normal;

            &:first-child {
                margin: 0 0 10px;
            }
        }
    }

    &__OND {
        &-container {
            display: flex;
            flex-direction: row;
        }
        &-icon {
            margin-right: 10px;
        }
        &-right-icon {
            margin-right: 15px; 
        }
        &-carrier-image {
            width: 60px;
            height: 30px;
        }
    }

    &__footer {
        display: flex;
        justify-content: space-between;
        align-items: flex-end;

        &-amount {
            p {
                margin: 0;
                // line-height: normal;
                &:first-child {
                    font-size: 20px;
                    font-weight: bold;
                }
            }
        }

        &-agent {
            color: #B0B0B1;
        }   

        button {
            padding: 5px 30px;
            background: #00ca6e;
            border-radius: 25px;
            color: #fff;
            border: 0;
            font-weight: 700;
            height: 35px;
            font-size: 1rem;
            margin-bottom: 8px;            
        }
    }
}