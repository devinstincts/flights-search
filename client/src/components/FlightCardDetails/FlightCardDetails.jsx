import React from 'react';
import PropTypes from 'prop-types';

import FlightLeg from '../FlightLeg/FlightLeg';
import AppPropTypes from '../../prop-types';
import './FlightCardDetails.css';

const FlightCardFooter = (itinerary, currencySymbol) => (
  <div className="card-footer-container">
    <div className="card-footer-amount-wrapper">
      <div className="card-footer-amount">
        {currencySymbol}
        {itinerary.PricingOptions[0].Price}
      </div>
      <div className="card-footer-agent">{itinerary.PricingOptions[0].Agents[0].Name}</div>
    </div>
    {/* eslint-disable-next-line react/button-has-type */}
    <button className="card-select-btn">
      Select
    </button>
  </div>
);

const FlightCardDetails = (props) => {
  const { itinerary, itinerary: { outboundLeg, inboundLeg }, currencySymbol } = props;
  return (
    <div className="flightCardDetailsContainer">
      <div>
        <FlightLeg flightLeg={outboundLeg} />
        <FlightLeg flightLeg={inboundLeg} />
      </div>
      {FlightCardFooter(itinerary, currencySymbol)}
    </div>
  );
};

FlightCardDetails.propTypes = {
  itinerary: AppPropTypes.Itinerary.isRequired,
  currencySymbol: PropTypes.string.isRequired,
};

export default FlightCardDetails;
