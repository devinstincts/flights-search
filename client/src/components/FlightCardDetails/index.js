import FlightCardDetails from './FlightCardDetails';

export default FlightCardDetails;
