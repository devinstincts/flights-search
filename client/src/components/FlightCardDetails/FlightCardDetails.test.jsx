import React from 'react';
import ReactDOM from 'react-dom';
import ShallowRenderer from 'react-test-renderer/shallow';
import FlightCardDetails from '.';
import mockRes from '../../mock.json';

describe('FlightCardDetails', () => {
  it('should render without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<FlightCardDetails
      itinerary={mockRes.Itineraries[0]}
      currencySymbol={mockRes.Currencies[0].Symbol}
    />, div);
  });

  it('snapshot', () => {
    const renderer = new ShallowRenderer();
    renderer.render(<FlightCardDetails
      itinerary={mockRes.Itineraries[0]}
      currencySymbol={mockRes.Currencies[0].Symbol}
    />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
