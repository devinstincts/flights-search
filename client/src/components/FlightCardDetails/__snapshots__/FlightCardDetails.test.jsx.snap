// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlightCardDetails snapshot 1`] = `
<div
  className="flightCardDetailsContainer"
>
  <div>
    <FlightLeg
      flightLeg={
        Object {
          "Arrival": "2019-10-15T14:35:00",
          "CarrierIds": Array [
            885,
            1033,
          ],
          "Carriers": Array [
            Object {
              "Code": "BE",
              "DisplayCode": "BE",
              "Id": 885,
              "ImageUrl": "https://s1.apideeplink.com/images/airlines/BE.png",
              "Name": "Flybe",
            },
            Object {
              "Code": "EI",
              "DisplayCode": "EI",
              "Id": 1033,
              "ImageUrl": "https://s1.apideeplink.com/images/airlines/EI.png",
              "Name": "Aer Lingus",
            },
          ],
          "Departure": "2019-10-15T07:20:00",
          "DestinationStation": Object {
            "Code": "LHR",
            "Id": 13554,
            "Name": "London Heathrow",
            "ParentId": 4698,
            "Type": "Airport",
          },
          "DestinationStationId": 13554,
          "Directionality": "Outbound",
          "Duration": 435,
          "FlightNumbers": Array [
            Object {
              "Carrier": Object {
                "Code": "BE",
                "DisplayCode": "BE",
                "Id": 885,
                "ImageUrl": "https://s1.apideeplink.com/images/airlines/BE.png",
                "Name": "Flybe",
              },
              "CarrierId": 885,
              "FlightNumber": "681",
            },
            Object {
              "Carrier": Object {
                "Code": "EI",
                "DisplayCode": "EI",
                "Id": 1033,
                "ImageUrl": "https://s1.apideeplink.com/images/airlines/EI.png",
                "Name": "Aer Lingus",
              },
              "CarrierId": 1033,
              "FlightNumber": "934",
            },
          ],
          "Id": "11235-1910150720--32302,-32753-1-13554-1910151435",
          "JourneyMode": "Flight",
          "OperatingCarrierIds": Array [
            885,
            1033,
          ],
          "OperatingCarriers": Array [
            Object {
              "Code": "BE",
              "DisplayCode": "BE",
              "Id": 885,
              "ImageUrl": "https://s1.apideeplink.com/images/airlines/BE.png",
              "Name": "Flybe",
            },
            Object {
              "Code": "EI",
              "DisplayCode": "EI",
              "Id": 1033,
              "ImageUrl": "https://s1.apideeplink.com/images/airlines/EI.png",
              "Name": "Aer Lingus",
            },
          ],
          "OriginStation": Object {
            "Code": "EDI",
            "Id": 11235,
            "Name": "Edinburgh",
            "ParentId": 2343,
            "Type": "Airport",
          },
          "OriginStationId": 11235,
          "SegmentIds": Array [
            0,
            1,
          ],
          "Segments": Array [
            Object {
              "ArrivalDateTime": "2019-10-15T08:15:00",
              "Carrier": Object {
                "Code": "BE",
                "DisplayCode": "BE",
                "Id": 885,
                "ImageUrl": "https://s1.apideeplink.com/images/airlines/BE.png",
                "Name": "Flybe",
              },
              "CarrierId": 885,
              "DepartureDateTime": "2019-10-15T07:20:00",
              "DestinationStation": Object {
                "Code": "BHD",
                "Id": 9889,
                "Name": "Belfast City",
                "ParentId": 876,
                "Type": "Airport",
              },
              "DestinationStationId": 9889,
              "Directionality": "Outbound",
              "Duration": 55,
              "FlightNumber": "681",
              "Id": 0,
              "JourneyMode": "Flight",
              "OperatingCarrier": Object {
                "Code": "BE",
                "DisplayCode": "BE",
                "Id": 885,
                "ImageUrl": "https://s1.apideeplink.com/images/airlines/BE.png",
                "Name": "Flybe",
              },
              "OperatingCarrierId": 885,
              "OriginStation": Object {
                "Code": "EDI",
                "Id": 11235,
                "Name": "Edinburgh",
                "ParentId": 2343,
                "Type": "Airport",
              },
              "OriginStationId": 11235,
            },
            Object {
              "ArrivalDateTime": "2019-10-15T14:35:00",
              "Carrier": Object {
                "Code": "EI",
                "DisplayCode": "EI",
                "Id": 1033,
                "ImageUrl": "https://s1.apideeplink.com/images/airlines/EI.png",
                "Name": "Aer Lingus",
              },
              "CarrierId": 1033,
              "DepartureDateTime": "2019-10-15T13:10:00",
              "DestinationStation": Object {
                "Code": "LHR",
                "Id": 13554,
                "Name": "London Heathrow",
                "ParentId": 4698,
                "Type": "Airport",
              },
              "DestinationStationId": 13554,
              "Directionality": "Outbound",
              "Duration": 85,
              "FlightNumber": "934",
              "Id": 1,
              "JourneyMode": "Flight",
              "OperatingCarrier": Object {
                "Code": "EI",
                "DisplayCode": "EI",
                "Id": 1033,
                "ImageUrl": "https://s1.apideeplink.com/images/airlines/EI.png",
                "Name": "Aer Lingus",
              },
              "OperatingCarrierId": 1033,
              "OriginStation": Object {
                "Code": "BHD",
                "Id": 9889,
                "Name": "Belfast City",
                "ParentId": 876,
                "Type": "Airport",
              },
              "OriginStationId": 9889,
            },
          ],
          "StopIds": Array [
            9889,
          ],
          "Stops": Array [
            Object {
              "Code": "BHD",
              "Id": 9889,
              "Name": "Belfast City",
              "ParentId": 876,
              "Type": "Airport",
            },
          ],
        }
      }
    />
    <FlightLeg
      flightLeg={
        Object {
          "Arrival": "2019-10-16T20:30:00",
          "CarrierIds": Array [
            881,
          ],
          "Carriers": Array [
            Object {
              "Code": "BA",
              "DisplayCode": "BA",
              "Id": 881,
              "ImageUrl": "https://s1.apideeplink.com/images/airlines/BA.png",
              "Name": "British Airways",
            },
          ],
          "Departure": "2019-10-16T19:05:00",
          "DestinationStation": Object {
            "Code": "EDI",
            "Id": 11235,
            "Name": "Edinburgh",
            "ParentId": 2343,
            "Type": "Airport",
          },
          "DestinationStationId": 11235,
          "Directionality": "Inbound",
          "Duration": 85,
          "FlightNumbers": Array [
            Object {
              "Carrier": Object {
                "Code": "BA",
                "DisplayCode": "BA",
                "Id": 881,
                "ImageUrl": "https://s1.apideeplink.com/images/airlines/BA.png",
                "Name": "British Airways",
              },
              "CarrierId": 881,
              "FlightNumber": "1458",
            },
          ],
          "Id": "13554-1910161905--32480-0-11235-1910162030",
          "JourneyMode": "Flight",
          "OperatingCarrierIds": Array [
            881,
          ],
          "OperatingCarriers": Array [
            Object {
              "Code": "BA",
              "DisplayCode": "BA",
              "Id": 881,
              "ImageUrl": "https://s1.apideeplink.com/images/airlines/BA.png",
              "Name": "British Airways",
            },
          ],
          "OriginStation": Object {
            "Code": "LHR",
            "Id": 13554,
            "Name": "London Heathrow",
            "ParentId": 4698,
            "Type": "Airport",
          },
          "OriginStationId": 13554,
          "SegmentIds": Array [
            2,
          ],
          "Segments": Array [
            Object {
              "ArrivalDateTime": "2019-10-16T20:30:00",
              "Carrier": Object {
                "Code": "BA",
                "DisplayCode": "BA",
                "Id": 881,
                "ImageUrl": "https://s1.apideeplink.com/images/airlines/BA.png",
                "Name": "British Airways",
              },
              "CarrierId": 881,
              "DepartureDateTime": "2019-10-16T19:05:00",
              "DestinationStation": Object {
                "Code": "EDI",
                "Id": 11235,
                "Name": "Edinburgh",
                "ParentId": 2343,
                "Type": "Airport",
              },
              "DestinationStationId": 11235,
              "Directionality": "Inbound",
              "Duration": 85,
              "FlightNumber": "1458",
              "Id": 2,
              "JourneyMode": "Flight",
              "OperatingCarrier": Object {
                "Code": "BA",
                "DisplayCode": "BA",
                "Id": 881,
                "ImageUrl": "https://s1.apideeplink.com/images/airlines/BA.png",
                "Name": "British Airways",
              },
              "OperatingCarrierId": 881,
              "OriginStation": Object {
                "Code": "LHR",
                "Id": 13554,
                "Name": "London Heathrow",
                "ParentId": 4698,
                "Type": "Airport",
              },
              "OriginStationId": 13554,
            },
          ],
          "StopIds": Array [],
          "Stops": Array [],
        }
      }
    />
  </div>
  <div
    className="card-footer-container"
  >
    <div
      className="card-footer-amount-wrapper"
    >
      <div
        className="card-footer-amount"
      >
        £
        502.38
      </div>
      <div
        className="card-footer-agent"
      >
        ebookers
      </div>
    </div>
    <button
      className="card-select-btn"
    >
      Select
    </button>
  </div>
</div>
`;
